### PR TITLE
#BN-19 feat: Guest/Buyer should be able to add items to their cart

### PR DESCRIPTION
**Pull request by Illustre13**
# Trello Board Link

- https://trello.com/c/4gsEwhLl

# PR Description
This PR will allow buyers to add items/product to the cart
# Description of tasks that were expected to be completed.
The backend needs to handle the addition of items to the cart, including storing the cart in the buyer's account or session and calculating the new cart total.
- The cart can be either be a new cart or the existing cart
GIVEN a post request to add an item to the cart
- WHEN the requesting user/token is a buyer and the request body is validated
- [x] The item should be added to the buyer's cart
- [x] The product details (name, price, image) should be retrieved from the database
- [x] The new cart total should be calculated and saved to the buyer's account or session
- [x] A confirmation message should be sent to the frontend to be displayed to the buyer
WHEN the request body fails any validation checks or the token is not of the role buyer
- [x] The appropriate 4XX response message should be sent to the frontend describing the validation error
# Description of tasks that were expected to be completed and that are not yet completed
- [x] None
# How this may be tested?
1. Make sure you are having a local postgres database and server running on port 8080
2. Make sure you are having `ecommerce-app-cogito-db-dev` database or create a new one
3. Run `make migrate` to migrate all tables to your `ecommerce-app-cogito-app-dv` database
4. Run `make seed` to copy the dummy data to your tables in db
6. In Postman or Insomnia create request with `http://localhost:9090/products/cart/add/{product_id}` with a GET method.
7. With Swagger documentation, Go to `http://localhost:9090/api-docs/#/Products/cart/add/{product_id}` and pass in the product id. You should first be logged in as the buyer in the system.
N.B: For running not using Docker, do `npm run migrate` , `npm run seed` and finally `npm run dev`
# Screenshots (If appropriate/applicable)
- [x] None
# Please check this Checklist before you submit your PR:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented on my code, particularly in hard-to-understand areas
- [x] My code generates no warnings
- [x] There are no vulnerabilities

### CODE REVIEW

File: src/database/models/cart.js [hunk 1]
```diff
@@ -13,10 +13,9 @@ module.exports = (sequelize, DataTypes) => {
   }
   cart.init(
     {
-      user_id: DataTypes.STRING,
-      product_id: DataTypes.STRING,
+      userId: DataTypes.STRING,
```
Comment (EvelyneUmubyeyi) [inline src/database/models/cart.js]: user_id should be an integer

File: src/database/migrations/20230327061757-create-cart.js [hunk 2]
```diff
@@ -8,10 +8,10 @@ module.exports = {
         primaryKey: true,
         type: Sequelize.INTEGER,
       },
-      user_id: {
+      userId: {
         type: Sequelize.STRING,
```
Comment (EvelyneUmubyeyi) [inline src/database/migrations/20230327061757-create-cart.js]: should be an integer

**Review by EvelyneUmubyeyi — CHANGES_REQUESTED** [review]

File: src/database/migrations/20230327061757-create-cart.js [hunk 3]
```diff
@@ -8,10 +8,10 @@ module.exports = {
...
       },
-      product_id: {
+      productId: {
         type: Sequelize.STRING,
```
Comment (EvelyneUmubyeyi) [inline src/database/migrations/20230327061757-create-cart.js]: should be an integer

File: src/database/models/cart.js [hunk 4]
```diff
@@ -13,10 +13,9 @@ module.exports = (sequelize, DataTypes) => {
...
+      productId: DataTypes.STRING,
```
Comment (EvelyneUmubyeyi) [inline src/database/models/cart.js]: should be an integer

**Review by EvelyneUmubyeyi — CHANGES_REQUESTED** [review]

File: src/database/models/cart.js (hunk 1 above)
Comment (Illustre13) [reply src/database/models/cart.js]: Yes, Changed it to integer!

File: src/database/migrations/20230327061757-create-cart.js (hunk 2 above)
Comment (Illustre13) [reply src/database/migrations/20230327061757-create-cart.js]: Have to create another migration to alter table carts with user_id of integer data type.

File: src/database/migrations/20230327061757-create-cart.js (hunk 3 above)
Comment (Illustre13) [reply src/database/migrations/20230327061757-create-cart.js]: Have to create another migration to alter table carts with product_id of integer data type.

File: src/database/models/cart.js (hunk 4 above)
Comment (Illustre13) [reply src/database/models/cart.js]: Have changed this to integer.

**Review by kunda4 — DISMISSED** [review]
Well done

**Review by EvelyneUmubyeyi — DISMISSED** [review]

**Review by kunda4 — DISMISSED** [review]
Well done

**Review by EvelyneUmubyeyi — DISMISSED** [review]

**Review by Niyobuzima — DISMISSED** [review]

**Review by EvelyneUmubyeyi — APPROVED** [review]

**Review by Niyobuzima — APPROVED** [review]

**Review by dushimeemma — APPROVED** [review]
Good job 🥇

**Pull request merged**